src/components/Products/CheckoutModal.js

```diff
@@ -8,64 +8,53 @@ class CheckoutModal extends React.Component {
         super(props);  
         this.itemsPurchased = this.itemsPurchased.bind(this);
         this.purchasesRef = Fire.database().ref(`user/${this.props.propId}/purchase`);
+        this.GetCart = this.GetCart.bind(this);
         this.state = {
             currentCart:[],
             totalPrice:0,
             redirect: false,
         }
     }
-
+    
     componentWillMount() {
         this.props.onRef(this)
-        var prices = 0;
-        const cartItems = [];
-        Fire.database().ref(`user/${this.props.propId}/cart`).on('value',function (snapshot){
-            let items = snapshot.val();
-            for(let id in items){
-                cartItems.push({id,...items[id]});
-                prices = parseFloat(prices) + (parseFloat(items[id].price) * parseFloat(items[id].quantity));
-            }
-        });
-        prices = prices.toFixed(2);
-        this.setState({
-            currentCart: {cartItems}
-        });
+        Fire.database().ref(`user/${this.props.propId}/cart`).on('value', this.GetCart.bind(this));
+    }
+    
+    componentWillUnmount() {
+        Fire.database().ref(`user/${this.props.propId}/cart`).off('value', this.GetCart.bind(this));
+    }
+    
+    GetCart(snapshot) {
+        let items = snapshot.val();
+        let prices = 0;
+        for(let item in items){
+            prices += (parseFloat(items[item].price) * parseInt(items[item].quantity));
+        }
         this.setState({
+            currentCart: items,
             totalPrice: prices
         });
     }
-
+    
     async itemsPurchased(){
         if(this.state.totalPrice == 0){
             alert("Your cart is empty")
-        }
+           }
         else{
-            let cartObj = {};
-            console.log(this.state.currentCart)
-            for(let i = 0; i <= this.state.currentCart; i++) {
-                let cart = this.state.currentCart[i];
-                cartObj[cart.id] = {
-                    quantity: cart.quantity,
-                    price: cart.price,
-                    productName: cart.productName,
-                    imageURL: cart.imageURL,
-                    discount: cart.discount
-                };
-            }
-            console.log("cartobg"+cartObj)
             let purchasedCart = {
-                cart: cartObj,
+                cart: this.state.currentCart,
                 timeStamp: Date.now(),
                 totalPrice: this.state.totalPrice,
                 userID:this.props.propId
             }
-            console.log("purchased cart"+purchasedCart)
+            console.log(purchasedCart)
             //add to purchase
             let userPurchases= (await Fire.database().ref(`user/${this.props.propId}`).once('value')).val().purchase;
-            let purchaseRef = Fire.database().ref(`purchase`).push(purchasedCart)
+            let purchaseRef = Fire.database().ref(`user/${this.props.propId}/purchase`).push(purchasedCart)
```

Comment:
This is wrong. use 
let purchaseRef = Fire.database().ref(`purchase`).push(purchasedCart)